Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Profile.hs

```diff
@@ -130,7 +143,33 @@ updatePerson personId req = do
       req.gender
       req.clientVersion
       req.bundleVersion
+  when (req.hasDisability == Just True && req.disability == Nothing) $ throwError $ InvalidRequest "Field disability can't be null if hasDisability is True"
```

Comment:
1. You can just have disability flag in update req instead of both
2. Update HasDisability field in the above update

Something like this

   ```
req.bundleVersion
if isJust req.disability then Just True else Just False

-- updatePerson part ends

whenJust req.disability $ \disability -> do
    customerDisability <- B.runInReplica $ PDisability.findByPersonId personId
    case customerDisability of
      Nothing -> do
        newDisability <- mkDisability disability -- generate id in mkDisability function
        PDisability.create newDisability
      Just disability -> PDisability.updateDisabilityByPersonId personId disabilityId disability.tag disability.description
```

Reply:
@0utkarsh

Reply:
1. req.disability is not of type Maybe Bool

Reply:
It is for `hasDisability` flag, not `req.disability`. You are checking if disability is sent in the req, on the basis of which you are updating the `hasDisability` flag. You can write your update function this way

updatePersonalInfo (Id personId) mbFirstName mbMiddleName mbLastName mbReferralCode mbEncEmail mbDeviceToken mbNotificationToken mbLanguage mbGender mbCVersion mbBVersion mbHasDisability = do
  now <- getCurrentTime
  let mbEmailEncrypted = mbEncEmail <&> unEncrypted . (.encrypted)
  let mbEmailHash = mbEncEmail <&> (.hash)
  updateWithKV
    ( [Se.Set BeamP.updatedAt now]
        <> [Se.Set BeamP.firstName mbFirstName | isJust mbFirstName]
        <> [Se.Set BeamP.middleName mbMiddleName | isJust mbMiddleName]
        <> [Se.Set BeamP.lastName mbLastName | isJust mbLastName]
        <> [Se.Set BeamP.emailEncrypted mbEmailEncrypted | isJust mbEmailEncrypted]
        <> [Se.Set BeamP.emailHash mbEmailHash | isJust mbEmailHash]
        <> [Se.Set BeamP.deviceToken mbDeviceToken | isJust mbDeviceToken]
        <> [Se.Set BeamP.notificationToken mbNotificationToken | isJust mbNotificationToken]
        <> [Se.Set BeamP.referralCode mbReferralCode | isJust mbReferralCode]
        <> [Se.Set BeamP.referredAt (Just now) | isJust mbReferralCode]
        <> [Se.Set BeamP.language mbLanguage | isJust mbLanguage]
        <> [Se.Set BeamP.gender (fromJust mbGender) | isJust mbGender]
        <> [Se.Set BeamP.clientVersion (versionToText <$> mbCVersion) | isJust mbCVersion]
        <> ([Se.Set BeamP.bundleVersion $ versionToText <$> mbBVersion | isJust mbBVersion])
        <> [Se.Set BeamP.hasDisability mbHasDisability | isJust mbHasDisability]
    )
    [Se.Is BeamP.id (Se.Eq personId)]
```

Reply:
But we can't do like this because, `hasDisability` field will be used in 3 use case 
1. Nothing -> existing user.
2. Just True -> People who recognise themself as disability.
3. Just False -> People who confirm that they don't have disability.

If we consider only `Just` and `Nothing` case , We will lose the data who chooses that they don't have disability.
We might nudge person who is opting **no disability** again and again.